commons/src/main/java/org/dhis2/commons/dialogs/MediaDialog.kt

```diff
@@ -0,0 +1,341 @@
+                                    url = audioUrl,
+                                    duration = generateRandomDuration(), //Todo: use object field instead
+                                    onClickMediaItem = { url -> onMediaItemClicked.invoke(url) },
+                                    dateOfLastUpdate = generateRandomDate() //Todo: use object field instead
+                                )
+                            }
+                        }
+                    }
+
+                    ButtonClose(
+                        modifier = Modifier.align(Alignment.End),
+                        onClick = { onDismiss.invoke() }
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun ButtonClose(
+    modifier: Modifier,
+    onClick: () -> Unit,
+) {
+    OutlinedButton(
+        onClick = onClick,
+        border = BorderStroke(0.5.dp, colorResource(R.color.colorPrimary_b0b)),
+        shape = RoundedCornerShape(50),
+        modifier = modifier
+    ) {
+        Text(
+            text = stringResource(id = R.string.media_dialog_label_close),
+            fontSize = 12.sp,
+            color = colorResource(R.color.textPrimary)
+        )
+    }
+}
+
+@Composable
+private fun MediaData(
+    title: String,
+    url: String,
+    duration: String,
+    dateOfLastUpdate: String,
+    onClickMediaItem: (url: String) -> Unit,
+) {
+    val textSize = remember { 12.sp }
+    val labelDuration = stringResource(R.string.media_dialog_label_duration)
+    val labelUpdate = stringResource(R.string.media_dialog_label_update)
+
+    Column(Modifier.clickable { onClickMediaItem.invoke(url) }) {
+        Text(
+            text = title,
+            fontSize = textSize,
+            fontWeight = FontWeight.Bold,
+        )
+        Row {
+            Text(
+                text = "$labelDuration:$SPACE_STRING",
+                fontSize = textSize,
+                fontWeight = FontWeight.Bold,
+            )
+            Text(
+                text = duration,
+                fontSize = textSize,
+            )
+        }
+        Row {
+            Text(
+                text = "$labelUpdate:$SPACE_STRING",
+                fontSize = textSize,
+                fontWeight = FontWeight.Bold,
+            )
+            Text(
+                text = dateOfLastUpdate,
+                fontSize = textSize,
+            )
+        }
+    }
+}
+
+@Composable
+private fun MediaDialogItemIcon(
+    type: MediaType,
+    size: Dp,
+) {
+    val mediaDrawableIcon = if (MediaType.VIDEO == type) {
+        R.drawable.media_dialog_video
+    } else {
+        R.drawable.media_dialog_audio
+    }
+    Image(
+        painter = painterResource(mediaDrawableIcon),
+        contentDescription = null,
+        contentScale = ContentScale.Crop,
+        alignment = Alignment.Center,
+        modifier = Modifier.size(size)
+    )
+}
+
+@Composable
+private fun MediaDialogMainIcon() {
```

Comment:
you can turn this into a generic icon component instead of repeating this in each function.

1. You should set params to this function: the vars you've declared can be the params of your function;
2. Since you already use the same props for all icon components, you can keep them and use all variable properties as parameters of your function